wabbit_tools.py

```diff
@@ -67,904 +29,391 @@ def err( msg ):
-    if (wavelet == "CDF44" or wavelet == "CDF62") and g<7:
-        warn("Not enough ghost nodes for wavelet %s g=%i < 7" % (wavelet, g) )
-    if (wavelet == "CDF40") and g<4:
-        warn("Not enough ghost nodes for wavelet %s g=%i < 4" % (wavelet, g) )
-        
-   
-    if time_step_method == "RungeKuttaChebychev":
-        if CFL_eta < 999:
-            warn('are you sure you did not forget to adjustl CFL_eta for the RKC scheme???')
-        if CFL_nu < 999:
-            warn('are you sure you did not forget to adjustl CFL_nu for the RKC scheme???')
-        if CFL != 0.75:
-            warn('are you sure you did not forget to adjustl CFL for the RKC scheme??? often we used 0.75.')    
-            
-    if time_step_method == "RungeKuttaGeneric":
-        if CFL_eta > 1.0:
-            warn('are you sure you did not forget to adjustl CFL_eta for the RK scheme? it may be unstable.')
-        if CFL_nu > 0.99*2.79/(float(dim)*np.pi**2):
-            warn('are you sure you did not forget to adjustl CFL_nu for the RK scheme? it may be unstable.')
-        if CFL >  1.0:
-            warn('are you sure you did not forget to adjustl CFL for the RK scheme? it may be unstable.')    
-            
-    # if somebody modifies the standard parameter file, users have to update their
-    # ini files they use. this is often forgoten and obnoxious. Hence, if we find
-    # value sthat no longer exist, warn the user.
-    if exists_ini_parameter( file, "Blocks", "number_data_fields" ) :
-        warn('Found deprecated parameter: [Blocks]::number_data_fields')
 
-    if exists_ini_parameter( file, "Physics", "initial_cond" ) :
-        warn('Found deprecated parameter: [Physics]::initial_cond')
 
-    if exists_ini_parameter( file, "Dimensionality", "dim" ) :
-        warn('Found deprecated parameter: [Dimensionality]::dim')
+#------------------------------------------------------------------------------
+# compute some keyvalues to compare the files
+#------------------------------------------------------------------------------
+def keyvalues(x0, dx, data):
+    max1, min1 = np.max(data), np.min(data)
+    mean1, squares1 = 0.0, 0.0
+    
+    # loop over all blocks
+    for i in range( data.shape[0] ):
+        if len(data.shape) == 3: ## 2D
+            mean1 = mean1 + np.product(dx[i,:]) * np.sum(data[i,:,:])
+            squares1 = squares1 + np.product(dx[i,:]) * np.sum(data[i,:,:]**2)
+        else: ## 3D
+            mean1 = mean1 + np.product(dx[i,:]) * np.sum(data[i,:,:,:])
+            squares1 = squares1 + np.product(dx[i,:]) * np.sum(data[i,:,:,:]**2)
+        
+    return(max1, min1, mean1, squares1 )
+
+
+# this object contains all important details about a read in wabbit-file
+# everything is neat and at one position and simplifies things
+# A grid is uniquely defined by its dimension, block size, domain size
+# The individual grid partition is uniquely defined by the number of blocks, treecode and level arrays
+class WabbitState:
```

Comment:
If I get it right, this is the content of a WABBIT HDF5 file, thus I find the name `WabbitState` confusing, because I associate it with a simulation state (running/failed). Why not call it `WabbitH5File` ?